test/Npgsql.Tests/Types/CompositeTests.cs

```diff
@@ -31,6 +31,56 @@ public async Task Basic()
             npgsqlDbType: null);
     }
 
+    [Test]
+    public async Task Basic_with_custom_default_translator()
+    {
+        await using var adminConnection = await OpenConnectionAsync();
+        var type = await GetTempTypeName(adminConnection);
+
+        await adminConnection.ExecuteNonQueryAsync($"CREATE TYPE {type} AS (x int, s text)");
+
+        var dataSourceBuilder = CreateDataSourceBuilder();
```

Comment:
Shouldn't we also test with slim just for the completion?

Reply:
Technically yes (and I considered it) it'd just be more code.